### ISSUE TITLE
NodeJSEnv does not propagate error output from Node.js

### ISSUE THREAD

**Issue by swachter** [scala-js/scala-js#3528]
Setting: I use ScalaJS to generate an ESModule that is used in an Angular application. The ESModule imports functionality from rxjs.

Test execution using the NodeJSEnv fails because the rxjs module can not be found. Unfortunately that problem is not visible when the tests are run from within sbt:

```
[debug] Loading JSEnv with linked file C:\Users\useswachte\IdeaProjects\katana-flow\scalajs\target\scala-2.12\sharedmodule-test-fastopt.mjs 
[debug] Starting process: node --experimental-modules                                                                                       
(node:11012) ExperimentalWarning: The ESM module loader is experimental.                                                                    
[error] org.scalajs.jsenv.ExternalJSEnv$NonZeroExitException: Node.js exited with code 1                                                    
[error]         at org.scalajs.jsenv.ExternalJSEnv$AbstractExtRunner.waitForVM(ExternalJSEnv.scala:126)                                     
[error]         at org.scalajs.jsenv.ExternalJSEnv$AsyncExtRunner$$anon$1.$anonfun$run$2(ExternalJSEnv.scala:191)                           
[error]         at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)                                                   
[error]         at scala.util.Try$.apply(Try.scala:209)                                                                                     
[error]         at org.scalajs.jsenv.ExternalJSEnv$AsyncExtRunner$$anon$1.run(ExternalJSEnv.scala:191)                                      
[error] (sharedModule / Test / loadedTestFrameworks) org.scalajs.jsenv.ExternalJSEnv$NonZeroExitException: Node.js exited with code 1       
[error] Total time: 9 s, completed 08.01.2019 16:46:33                                                                                      
```

When the test is run on the command line the reason why Node.js exited with code 1 is revealed:

```
$ node --experimental-modules sharedmodule-test-fastopt.mjs                        
(node:15132) ExperimentalWarning: The ESM module loader is experimental.           
{ Error: Cannot find module rxjs/Rx                                                
    at search (internal/modules/esm/default_resolve.js:29:12)                      
    at Loader.resolve [as _resolve] (internal/modules/esm/default_resolve.js:65:11)
    at Loader.resolve (internal/modules/esm/loader.js:58:33)                       
    at Loader.getModuleJob (internal/modules/esm/loader.js:113:40)                 
    at ModuleWrap.promises.module.link (internal/modules/esm/module_job.js:32:40)  
    at link (internal/modules/esm/module_job.js:31:36) code: 'MODULE_NOT_FOUND' }  
```

**Comment by swachter** [comment]
A standalone reproduction will take some time. I looked into `ExternalJSEnv` and saw that only the error stream of the node process is redirected:

```
    protected def startVM(): Process = {
      val vmArgs = getVMArgs()
      val vmEnv  = getVMEnv()

      val allArgs = executable +: vmArgs
      val pBuilder = new ProcessBuilder(allArgs: _*)
      pBuilder.redirectErrorStream(true) // merge stderr into stdout

      pBuilder.environment().clear()
      for ((name, value) <- vmEnv)
        pBuilder.environment().put(name, value)

      logger.debug("Starting process: " + allArgs.mkString(" "))

      pBuilder.start()
    }
```

Maybe redirecting the output stream (at least in debug mode) would fix the problem.

**Comment by sjrd** [comment]
The redirection only means that stderr and stdout coming from Node.js will be merged into a single output from the host JVM point of view (i.e., sbt). The merged stream is already sent to the host JVM's stdout. So we're already doing what you suggest.

In Scala.js 1.x we don't redirect anything, and instead Node.js stdout (resp. stderr) is directly sent to the host JVM's stdout (resp. stderr). That removes one level of indirection, but mostly that shouldn't make any visible difference when sbt is run in a terminal (as opposed to within another application which itself redirects stdout/stderr).

**Comment by sjrd** [comment]
Thanks for the reproduction.

Minimized down to:
```scala
package abc

import scala.scalajs.js
import scala.scalajs.js.annotation._

@js.native
@JSImport("doesnotexist.mjs", "Foo")
object Foo extends js.Any

object Adder {
  def main(args: Array[String]): Unit = {
    println(Foo)
  }
}
```
with `run`.

Reproducible with 0.6.x, but not with 1.x. So that's one thing, at least 1.x correctly displays the error message (in `run` and `test`).

**Comment by sjrd** [comment]
Fixed in 12a7988b381acc5aeaed7b6c1bcf1922ef242a77.
